provDetective.py

```diff
@@ -0,0 +1,99 @@
+
+import MySQLdb
+
+class ProvDetective(object):
+    """
+    This class is helping to poke around inside provenance to find things.
+    """
+    def __init__(self, host, port, user, password, db):
+        self._conn = MySQLdb.connect(host=host, port=port, user=user,
+                                     passwd=password, db=db)
+        self._cursor = self._conn.cursor()
+
+    def __del__(self):
+        self._conn.close()
+
+    def _getRandomObjectId(self):
+        self._cursor.execute('SELECT objectId FROM Object ORDER BY RAND() LIMIT 1')
+        row = self._cursor.fetchone()
+        return row[0]
+
+    def _getRandomSourceId(self):
+        self._cursor.execute('SELECT sourceId FROM Source ORDER BY RAND() LIMIT 1')
+        row = self._cursor.fetchone()
+        return row[0]
+
+    def nodesThatProcessedObject(self, objectId=None):
+        """
+        Prints which node processed a given object. If objectId is none, the
+        function will do it for one randomly selected object.
+        """
+        if objectId is None:
+            objectId = self._getRandomObjectId()
+
+        # find all exposures that have sources corresponding to this object
+        self._cursor.execute('''
+SELECT scExposureId, sourceId, sceGroupId, taskExecId, taskName, nodeName
+FROM   Source
+JOIN   ScienceCalibratedExposure USING(scExposureId)
+JOIN   prv_SCEExposureToGroup USING(scExposureId)
+JOIN   prv_TaskExecutionToSCEGroup USING(sceGroupId)
+JOIN   prv_TaskExecution USING(taskExecId)
+JOIN   prv_Node USING(nodeId)
+JOIN   prv_Task USING(taskId)
+WHERE  objectId=%s''' % objectId)
```

Comment:
This looks like it mixes qserv tables with provenance tables, does it mean that provenance schema has to be in qserv?